bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/HistoryStore.java

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vector Informatik GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Vector Informatik GmbH - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.ui.internal.findandreplace;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.Vector;
+
+import org.eclipse.jface.dialogs.IDialogSettings;
+
+/**
+ * Store for search-histories or replace-histories inside of a dialog.. Stores
+ * the nodes using the DialogSettings-mechanism.
+ */
+public class HistoryStore {
+	IDialogSettings settingsManager;
+	private int historySize;
+	Vector<String> history;
+	String sectionName;
+
+	/**
+	 * @param settingsManager
+	 * @param sectionName     The name of the section in the DialogSettings
+	 *                        containing the history
+	 * @param historySize     how many entries to keep in the history
+	 */
+	public HistoryStore(IDialogSettings settingsManager, String sectionName, int historySize) {
+		this.settingsManager = settingsManager;
+		this.historySize = historySize;
+		loadSection(sectionName);
+	}
+
+	public Vector<String> get() {
+		return history;
+	}
+
+	public String get(int index) {
+		return history.get(index);
+	}
+
+
+	public void add(String historyItem) {
+		if (sectionName == null) {
+			throw new Error("No section loaded"); //$NON-NLS-1$
+		}
+		if (historyItem != null && !historyItem.isEmpty())
+			history.add(0, historyItem);
+
+		writeHistory();
+	}
+
+	public boolean isEmpty() {
+		return history.isEmpty();
+	}
+
+	private void loadSection(String newSectionName) {
+		this.sectionName = newSectionName;
+
+		history = new Vector<>(Arrays.asList(settingsManager.getArray(newSectionName)));
+	}
+
+	/**
+	 * Writes the given history into the given dialog store.
+	 * @since 3.2
```

Comment:
there are still plenty since tags in this commit

Reply:
I will look through the code again, more thoroughly